spring-integration-core/src/main/java/org/springframework/integration/channel/MessageSourcePollableChannel.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.channel;
+
+import org.springframework.integration.core.MessageSource;
+import org.springframework.messaging.Message;
+
+/**
+ * A PollableChannel implementation that simply delegates to a {@link MessageSource}.
+ * This allows channel interceptor functionality (postReceive) to be applied to
+ * messages from any message source.
+ * Timeout is ignored.
+ *
+ * <p>
+ *
+ * ####### THIS IS JUST A TRIAL BALOON #######
+ *
+ * It's an easy way to apply interceptors to messages created by message sources which
+ * can be polled on an ad-hoc basis.
+ * But another option would be to inject an interceptor list into AbstractMessageSource
+ * itself.
+ *
+ * Thoughts?
```

Comment:
Looks like you try to repeat what we really have with the `SourcePollingChannelAdapter` and its `adviceChain()`.
That really doesn't look like functionality for `channel` abstraction: http://www.enterpriseintegrationpatterns.com/patterns/messaging/MessageChannel.html.

I don't have too much arguments against, but I wouldn't commit it into the `master` since it is not a design we follow in the Framework. Isn't it? 

It may live in the tests though, but we even once have deprecated and removed `MessageTransformingChannelInterceptor` although that is really possible...

Reply:
The problem is I need a way in SCSt to add interceptors to modify the message produced by the source before  the app gets the message and I am trying to avoid adding a bridge for that purpose (extra hop).

So I have two choices - add some kind of interceptor chain to the message source so that `receive()` returns a modified message, or use this technique and use channel interceptors to `postReceive()` the message.

In the first case, the app would have to use `source.receive()` in the second case he just gets a `PollableChannel` - which is closer to the current message-driven `@StreamListener`.

I suppose we could put this "dummy" channel in SCSt instead.